Analysis/Tutorials/src/histogramRegistry.cxx

```diff
@@ -294,14 +294,15 @@ struct GTask {
   }
 };
 
-WorkflowSpec defineDataProcessing(ConfigContext const&)
+WorkflowSpec defineDataProcessing(ConfigContext const& ctx)
 {
+  auto suffix = ctx.options().get<std::string>("workflow-suffix");
   return WorkflowSpec{
-    adaptAnalysisTask<ETask>("output-obj-test"),
-    adaptAnalysisTask<ATask>("eta-and-phi-histograms"),
-    adaptAnalysisTask<BTask>("filtered-histograms"),
-    adaptAnalysisTask<CTask>("dimension-test"),
-    adaptAnalysisTask<DTask>("realistic-example"),
-    adaptAnalysisTask<FTask>("tlist-test"),
-    adaptAnalysisTask<GTask>("configurables-test")};
+    adaptAnalysisTask<ETask>(("output-obj-test" + suffix).c_str()),
+    adaptAnalysisTask<ATask>(("eta-and-phi-histograms" + suffix).c_str()),
+    adaptAnalysisTask<BTask>(("filtered-histograms" + suffix).c_str()),
+    adaptAnalysisTask<CTask>(("dimension-test" + suffix).c_str()),
+    adaptAnalysisTask<DTask>(("realistic-example" + suffix).c_str()),
+    adaptAnalysisTask<FTask>(("tlist-test" + suffix).c_str()),
+    adaptAnalysisTask<GTask>(("configurables-test" + suffix).c_str())};
```

Comment:
Why not simply passing the `ctx` to the `adaptAnalysisTask`?

One possible solution we could have is to make the `adaptAnalysisTask()` function be a member of (an object returned by) the ConfigContext. Something like:

```c++
ctx.taskMaker().makeAnalysisTask<ETask>();
```

or alternatively have an helper method for the workflow, rather than for the simple task:

```c++
return adaptAnalysisWorkflow(ctx, TaskSpec{"output-obj-test"}, TaskSpec{"eta-and-phi-histograms"}});
```

What do you think?

Reply:
The other solution would be to inject the suffix after defineDataProcessing has been called, no?

Reply:
Another solution would be that `defineDataProcessing` does not return a `std::vector<DataProcessorSpec>`, but a `ConfigureDataProcessorAction` which then can be used to do the lazy configuration of the `DataProcessor`. I think we can even guarantee backward compatibility by having a constructor for `ConfigureDataProcessorAction(DataProcessorSpec)`.

Reply:
Hi,
true, we can just pass the ctx and append the suffix inside the `adaptAnalysisTask()`.

`ConfigureDataProcessorAction` - hm, still we need to pass the ctx somewhere, so it'll still mean a change in analysis workflows. I don't see how it could be much different from just passing the ctx to `adaptAnalysisTask()`.

`adaptAnalysisWorkflow` looks also good to me since ctx would be used just once. Though it is a bigger change on the user side ;)

Reply:
Regarding ConfigureDataProcessorAction: my idea would be that the context is passed outside the user code.

Basically we would go from `std::vector<DataProcessorSpec> runDataProcessing(ConfigContext...)` to `std::vector<ConfigureDataProcessorAction> runDataProcessing(ConfigContext...)`, where `ConfigureDataProcessorAction` would be something like:

```
enum struct DataProcessorAction : char {
  CopyFullFromProtoToTarget,         // The default action, copy the whole DataProcessorSpec to the one labelled as target
  AppendInputsFromProtoToTarget, // Take the inputs in proto and replace them in the one labelled.
  AppendOutputsFromProtoToTarget
};

struct ConfigureDataProcessorAction {
  std::string target;
  DataProcessorSpec *proto;
  DataProcessorAction action; 
};
```

Reply:
It seems to me it'd be then the same solution as your primary one, with appending the suffix after the tasks creation in `overrideSuffix()`.

We have direct access to output object only inside `adaptAnalysisTask()`. So, later it is not possible to change the task hashes inside objects and copying the spec doesn't help.

Maybe I am wrong and changing something in DataProcessorSpec will be enough to match the hashes properly in sinks...? Though I haven't suceeded with it previously. In that case, wouldn't it be simplier to stay with `overrideSuffix()`?